### PR TITLE
Fixes #33272 - implement dry-run for restore

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

File: definitions/scenarios/restore.rb [hunk 1]
```diff
@@ -6,19 +6,27 @@ class Restore < ForemanMaintain::Scenario
...
 
     # rubocop:disable Metrics/MethodLength,Metrics/AbcSize
+    # rubocop:disable Metrics/CyclomaticComplexity,Metrics/PerceivedComplexity
     def compose
       backup = ForemanMaintain::Utils::Backup.new(context.get(:backup_dir))
 
       add_steps(find_checks(:root_user))
       supported_version_check
       add_steps_with_context(Checks::Restore::ValidateBackup,
-                             Procedures::Restore::Confirmation,
```
Comment (evgeni) [inline definitions/scenarios/restore.rb:19]: The confirmation should really happen after *all* checks, this is unrelated to the `dry-run` case I think (can split out in a separate PR, if you want to?)

File: definitions/scenarios/restore.rb [hunk 2]
```diff
@@ -6,19 +6,27 @@ class Restore < ForemanMaintain::Scenario
...
                              Checks::Restore::ValidateHostname,
-                             Checks::Restore::ValidateInterfaces,
+                             Checks::Restore::ValidateInterfaces)
+
+      if context.get(:dry_run)
+        self.class.metadata[:run_strategy] = :fail_slow
+        return
+      end
```
Comment (evgeni) [inline definitions/scenarios/restore.rb:27]: Happy to see a "nicer" implementation of "if dry run, just add checks and fail_slow", but this one works, so :woman_shrugging:

**Comment by evgeni** [comment]
Example output with dry run failing:

```console
# foreman-maintain restore --dry-run /backup
Running Restore backup
================================================================================
Check if command is run as root user:                                 [OK]
--------------------------------------------------------------------------------
Validate backup has appropriate files:                                [OK]
--------------------------------------------------------------------------------
Validate hostname is the same as backup:                              [FAIL]
The hostname in the backup does not match the hostname of the system
--------------------------------------------------------------------------------
Validate network interfaces match the backup:                         [OK]
--------------------------------------------------------------------------------
Scenario [Restore backup] failed.

The following steps ended up in failing state:

  [restore-validate-hostname]

Resolve the failed steps and rerun
the command. In case the failures are false positives,
use --whitelist="restore-validate-hostname"
```

And non-dry-run failing:

```console
# foreman-maintain restore /backup
Running Restore backup
================================================================================
Check if command is run as root user:                                 [OK]
--------------------------------------------------------------------------------
Validate backup has appropriate files:                                [OK]
--------------------------------------------------------------------------------
Validate hostname is the same as backup:                              [FAIL]
The hostname in the backup does not match the hostname of the system
--------------------------------------------------------------------------------
Scenario [Restore backup] failed.

The following steps ended up in failing state:

  [restore-validate-hostname]

Resolve the failed steps and rerun
the command. In case the failures are false positives,
use --whitelist="restore-validate-hostname"



Running Rescue Restore backup
================================================================================

```

**Review by ehelms — APPROVED** [review]

**Comment by evgeni** [comment]
added some very basic tests :)

File: definitions/scenarios/restore.rb (hunk 1 above)
Comment (upadhyeammit) [reply definitions/scenarios/restore.rb:19]: Yep, confirmation after _all checks_ would be nice, +1 for separate pr.

File: definitions/scenarios/restore.rb (hunk 2 above)
Comment (upadhyeammit) [reply definitions/scenarios/restore.rb:27]: It works as expected so good to go from my side.

**Review by upadhyeammit — APPROVED** [review]
Works as expected, maybe we can have separate pr for the confirmation procedure but thats not the blocker.

**Comment by evgeni** [comment]
rebased!
